### PR TITLE
[FLINK-19181][python] Make python processes respect the calculated managed memory fraction

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request make python processes to respect the calculated managed memory fraction.*

## Brief change log

  - *Make python processes to respect the calculated managed memory fraction.*
  - *Share the Python process between the Python operators in the same slot*

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by HuangXingBo — COMMENTED** [review]
@dianfu Thanks a lot for the PR. Good Work! I have left only a minor comments.

**Comment by HuangXingBo** [comment]
@dianfu The memory configuration shown in the python udf related documents is as follows 
`table_env.get_config().get_configuration().set_string("taskmanager.memory.task.off-heap.size", '80m')`
which I think can be removed together.

**Comment by dianfu** [comment]
@HuangXingBo Thanks for the review. Updated the PR.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupAggregate.scala [hunk 1]
```diff
@@ -36,14 +37,12 @@ import org.apache.flink.table.planner.plan.rules.physical.batch.BatchExecJoinRul
 import org.apache.flink.table.planner.plan.utils.{FlinkRelOptUtil, RelExplainUtil}
 import org.apache.flink.table.runtime.typeutils.InternalTypeInfo
 import org.apache.flink.table.types.logical.RowType
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupAggregate.scala:39]: unnecessary change

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupAggregate.scala [hunk 2]
```diff
@@ -36,14 +37,12 @@ import org.apache.flink.table.planner.plan.rules.physical.batch.BatchExecJoinRul
...
 import org.apache.calcite.plan.{RelOptCluster, RelOptRule, RelTraitSet}
 import org.apache.calcite.rel.RelDistribution.Type.{HASH_DISTRIBUTED, SINGLETON}
 import org.apache.calcite.rel.`type`.RelDataType
 import org.apache.calcite.rel.core.AggregateCall
 import org.apache.calcite.rel.{RelCollationTraitDef, RelCollations, RelNode, RelWriter}
 import org.apache.calcite.util.{ImmutableIntList, Util}
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupAggregate.scala:46]: ditto

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonCalc.scala [hunk 3]
```diff
@@ -19,11 +19,10 @@
 package org.apache.flink.table.planner.plan.nodes.physical.batch
 
 import org.apache.flink.api.dag.Transformation
+import org.apache.flink.core.memory.ManagedMemoryUseCase
 import org.apache.flink.table.data.RowData
 import org.apache.flink.table.planner.delegation.BatchPlanner
 import org.apache.flink.table.planner.plan.nodes.common.CommonPythonCalc
-import org.apache.flink.table.planner.plan.nodes.exec.ExecNode
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonCalc.scala:26]: unnecessary change

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupWindowAggregate.scala [hunk 4]
```diff
@@ -41,7 +42,6 @@ import org.apache.flink.table.planner.plan.nodes.exec.{BatchExecNode, ExecNode}
 import org.apache.flink.table.planner.plan.nodes.physical.batch.BatchExecPythonGroupWindowAggregate.ARROW_PYTHON_GROUP_WINDOW_AGGREGATE_FUNCTION_OPERATOR_NAME
 import org.apache.flink.table.runtime.typeutils.InternalTypeInfo
 import org.apache.flink.table.types.logical.RowType
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonGroupWindowAggregate.scala:44]: unnecessary change

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonOverAggregate.scala [hunk 5]
```diff
@@ -28,12 +29,10 @@ import org.apache.flink.table.functions.python.PythonFunctionInfo
 import org.apache.flink.table.planner.calcite.FlinkTypeFactory
 import org.apache.flink.table.planner.delegation.BatchPlanner
 import org.apache.flink.table.planner.plan.nodes.common.CommonPythonAggregate
-import org.apache.flink.table.planner.plan.nodes.exec.ExecNode
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonOverAggregate.scala:31]: unnecessary change

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonOverAggregate.scala [hunk 6]
```diff
@@ -28,12 +29,10 @@ import org.apache.flink.table.functions.python.PythonFunctionInfo
...
 import org.apache.flink.table.planner.plan.nodes.physical.batch.BatchExecPythonOverAggregate.ARROW_PYTHON_OVER_WINDOW_AGGREGATE_FUNCTION_OPERATOR_NAME
 import org.apache.flink.table.planner.plan.utils.OverAggregateUtil.getLongBoundary
 import org.apache.flink.table.runtime.typeutils.InternalTypeInfo
 import org.apache.flink.table.types.logical.RowType
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchExecPythonOverAggregate.scala:36]: ditto

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamExecPythonGroupWindowAggregate.scala [hunk 7]
```diff
@@ -31,15 +32,13 @@ import org.apache.flink.table.planner.delegation.StreamPlanner
...
 import org.apache.flink.table.planner.plan.nodes.common.CommonPythonAggregate
-import org.apache.flink.table.planner.plan.nodes.exec.ExecNode
 import org.apache.flink.table.planner.plan.nodes.physical.stream.StreamExecPythonGroupWindowAggregate.ARROW_STREAM_PYTHON_GROUP_WINDOW_AGGREGATE_FUNCTION_OPERATOR_NAME
 import org.apache.flink.table.planner.plan.utils.AggregateUtil._
 import org.apache.flink.table.planner.plan.utils.{KeySelectorUtil, WindowEmitStrategy}
 import org.apache.flink.table.runtime.operators.window.assigners._
 import org.apache.flink.table.runtime.operators.window.triggers.{ElementTriggers, EventTimeTriggers, ProcessingTimeTriggers, Trigger}
 import org.apache.flink.table.runtime.typeutils.InternalTypeInfo
 import org.apache.flink.table.types.logical.RowType
-
```
Comment (HuangXingBo) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamExecPythonGroupWindowAggregate.scala:42]: unnecessary change

**Review by HuangXingBo — COMMENTED** [review]
@dianfu Thanks a lot for the update. I only left some comments about unnecessary changes for import.

**Comment by dianfu** [comment]
@HuangXingBo Updated.

**Comment by HuangXingBo** [comment]
@dianfu Thanks a lot for the update. LGTM

**Comment by dianfu** [comment]
@HuangXingBo Thanks for the review. Will merge once the tests passed.

**Comment by dianfu** [comment]
@flinkbot run azure

**Comment by dianfu** [comment]
@flinkbot run azure

**Comment by dianfu** [comment]
@flinkbot run azure
